thriftpy/thrift.py

```diff
@@ -192,9 +192,28 @@ def _negotiation(self):
             result.read(self._iprot)
             self._iprot.read_message_end()
 
-    def _send(self, _api, **kwargs):
+    def _req(self, *args, **kwargs):
```

Comment:
There are some concerns here.

1. Exception handling:
    1. Exception might be caused by network failure, hence further reading from the same socket will fail as well.
    2. Has rewritten trackback history, a minor one actually.
2. Status is acknowledged by exception.
    1. Has not distinguished what kind of exception, network? Or thrift defined exception.
3. Read header from server.
    We only care about one call, and don't need to construct the whole call chain at programming level. It will be collected at every call, and reconstructed offline by analyzing. Having every header in the original caller is a over kill.